nilearn/mass_univariate/tests/test_permuted_least_squares.py

```diff
@@ -422,284 +515,252 @@ def test_permuted_ols_withcovar(random_state=0):
+        output["t"], tested_var, target_var, confounding_vars
+    )
+    assert ref_score.shape == (n_regressors, n_descriptors)
+    assert output["t"].shape == (n_regressors, n_descriptors)
 
     # same thing but with model_intercept=True to check it has no effect
-    _, own_scores_intercept, _ = permuted_ols(
+    output_intercept = permuted_ols(
         tested_var,
         target_var,
         confounding_vars,
         model_intercept=True,
         n_perm=0,
         random_state=random_state,
+        output_type="dict",
+    )
+    compare_to_ref_score(
+        output_intercept["t"], tested_var, target_var, confounding_vars
     )
-    assert own_scores_intercept.shape == (n_regressors, n_descriptors)
-    assert_array_almost_equal(ref_scores, own_scores_intercept, decimal=6)
+    assert output_intercept["t"].shape == (n_regressors, n_descriptors)
 
 
-# Test one-sided versus two-sided
-def test_sided_test(random_state=0):
+def test_one_sided_versus_two_test(random_state=RANDOM_STATE):
     """Check that a positive effect is always better \
     recovered with one-sided."""
     rng = check_random_state(random_state)
 
-    # design parameters
-    n_samples = 50
+    n_samples = N_SAMPLES
     n_descriptors = 100
     n_regressors = 1
-
-    # create design
     target_var = rng.randn(n_samples, n_descriptors)
     tested_var = rng.randn(n_samples, n_regressors)
 
-    # permuted OLS
     # one-sided
-    neg_log_pvals_onesided, _, _ = permuted_ols(
+    output_1_sided = permuted_ols(
         tested_var,
         target_var,
         model_intercept=False,
         two_sided_test=False,
-        n_perm=100,
+        n_perm=N_PERM,
         random_state=random_state,
+        output_type="dict",
     )
-    assert neg_log_pvals_onesided.shape == (n_regressors, n_descriptors)
+    assert output_1_sided["logp_max_t"].shape == (n_regressors, n_descriptors)
 
     # two-sided
-    neg_log_pvals_twosided, _, _ = permuted_ols(
+    output_2_sided = permuted_ols(
         tested_var,
         target_var,
         model_intercept=False,
         two_sided_test=True,
-        n_perm=100,
+        n_perm=N_PERM,
         random_state=random_state,
-        verbose=1,
+        output_type="dict",
     )
-    assert neg_log_pvals_twosided.shape == (n_regressors, n_descriptors)
+    assert output_2_sided["logp_max_t"].shape == (n_regressors, n_descriptors)
 
-    positive_effect_location = neg_log_pvals_onesided > 1
+    positive_effect_location = output_1_sided["logp_max_t"] > 1
     assert_equal(
         np.sum(
-            neg_log_pvals_twosided[positive_effect_location]
-            - neg_log_pvals_onesided[positive_effect_location]
+            output_2_sided["logp_max_t"][positive_effect_location]
+            - output_1_sided["logp_max_t"][positive_effect_location]
             > 0
         ),
         0,
     )
 
 
-def test_sided_test2(random_state=0):
+def test_sided_test2(random_state=RANDOM_STATE):
```

Comment:
Based on what the docstring says, I'd go for a more informative function name:

```suggestion
def test_two_sided_recover_positive_and_negative_effects(random_state=RANDOM_STATE):
```